components/builder-protocol/src/originsrv.rs

```diff
@@ -1158,6 +1158,66 @@ impl Serialize for OriginProjectList {
     }
 }
 
+impl Serialize for OriginSecret {
+    fn serialize<S>(&self, serializer: S) -> result::Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        let mut state = serializer.serialize_struct("secret", 2)?;
```

Comment:
I think the number in this call refers to the number of fields being serialized in the struct, so I _think_ this should be 4.  I'm assuming everything serializes normally though, so honestly, I'm a little baffled.